firmware/init/sensor/init_vehicle_speed_sensor.cpp

```diff
@@ -2,25 +2,24 @@
 
 #include "init.h"
 #include "vehicle_speed.h"
-#include "function_pointer_sensor.h"
+#include "frequency_sensor.h"
+#include "vehicle_speed_converter.h"
+
+static FrequencySensor vehicleSpeedSensor(SensorType::VehicleSpeed, MS2NT(500));
+static VehicleSpeedConverter vehicleSpeedConverter;
 
-struct GetVehicleSpeedWrapper {
-	DECLARE_ENGINE_PTR;
 
-	float getVehicleSpeed() {
-		return ::getVehicleSpeed(PASS_ENGINE_PARAMETER_SIGNATURE);
-	}
-};
 
-static GetVehicleSpeedWrapper vehicleSpeedWrapper;
+void initVehicleSpeedSensor(DECLARE_CONFIG_PARAMETER_SIGNATURE) {
```

Comment:
you will need an `INJECT_ENGINE_REFERENCE(&vehicleSpeedConverter);` here somewhere

Reply:
Noted